### PR TITLE
[WIP] Remove the requirement for double update in forward

### PR DESCRIPTION
**Pull request by tchaton**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?

This PR optimizer the forward method of the Metric base class to prevent performing 2 updates.

Parts of https://github.com/PyTorchLightning/metrics/issues/344

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@tchaton any update here? :rabbit:

**Comment by Borda** [comment]
@tchaton any update here? :rabbit:

File: CHANGELOG.md [hunk 1]
```diff
@@ -82,6 +83,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Scalar metrics will now consistently have additional dimensions squeezed ([#622](https://github.com/PyTorchLightning/metrics/pull/622))
```
Comment (SkafteNicki) [inline CHANGELOG.md:86]:
```suggestion
```

**Comment by Borda** [comment]
@SkafteNicki, are we going to proceed with this one? :rabbit:

**Comment by SkafteNicki** [comment]
Closing in favour of https://github.com/PyTorchLightning/metrics/pull/984

**Pull request closed without merging**